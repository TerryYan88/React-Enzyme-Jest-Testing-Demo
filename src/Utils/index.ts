/* eslint-disable react/forbid-foreign-prop-types */
import {ShallowWrapper} from "enzyme";
import checkPropTypes from "check-prop-types";
import { applyMiddleware,createStore } from "redux";
import { rootReducer } from "../redux/reducers/rootReducer";


export const testStore = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware()(createStore)
    return createStoreWithMiddleware(rootReducer,initialState);
}


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}

export const checkProps = (component:any,expectedProps:{})=>{
    const propsErr = checkPropTypes(component.propTypes,expectedProps,'props',component.name);
    return propsErr;
}


