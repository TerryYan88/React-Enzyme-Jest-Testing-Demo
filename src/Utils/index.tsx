
import {ShallowWrapper} from "enzyme";


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}