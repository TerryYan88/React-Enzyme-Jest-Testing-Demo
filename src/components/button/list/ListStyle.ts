import styled from "styled-components";

export const ListLi = styled.li`
    li{

    }
`