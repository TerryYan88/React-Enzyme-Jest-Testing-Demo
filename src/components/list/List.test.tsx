import React from "react";
import renderer from "react-test-renderer";
import { shallow, ShallowWrapper, } from "enzyme";
import ListItem, { ListProps } from "./List";
import { Ul, Li } from "./ListStyle";
import { findByTestAttr, checkProps } from "../../Utils/index";


const setUp = (props: ListProps) => {
    const component = shallow(<ListItem {...props} />) // can test props 
    return component;
}

describe("List component should", () => {
    it("have a snapshot", () => {
        const props: ListProps = {
            list: [
                {
                    userId:"test 0",
                    id: "test 0",
                    title: "Example title",
                    body: "Some text"
                }
            ]
        };
        const component = (<ListItem {...props} />);
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    })
    describe("Checking PropTypes", () => {
        it("Should not throw a warning", () => {
            const expectedProps = {
                list: [
                    {
                        userId:"test 0",
                        id: "test 0",
                        title: "Example title",
                        body: "Some text"
                    }
                ]
            }
            const propsError = checkProps(ListItem, expectedProps)
            expect(propsError).toBeUndefined();
        })
    })
    describe("List component shoud", () => {
        let wrapper: ShallowWrapper
        beforeEach(() => {
            const props = {
                list: [
                    {
                        userId:"test 0",
                        id: "test 0",
                        title: "Example title",
                        body: "Some text"
                    }
                ]
            }
            wrapper = setUp(props)
        })
        it("render UL", () => {
            const UlContainer = wrapper.find(Ul);
            expect(UlContainer.length).toBe(1);
            const ulComponent = findByTestAttr(wrapper, "ulComponent");
            expect(ulComponent.length).toBe(1);
        })
        it("render li", () => {
            const LiContainer = wrapper.find(Li);
            expect(LiContainer.length).toBe(1);
            const liComponent = findByTestAttr(wrapper, "liComponent");
            expect(liComponent.length).toBe(1);
        })
        it("render title", () => {
            const componentTitle = findByTestAttr(wrapper, "componentTitle");
            expect(componentTitle.length).toBe(1);
        })
        it("render desc", () => {
            const componentDesc = findByTestAttr(wrapper, "componentDesc");
            expect(componentDesc.length).toBe(1);
        })
    })
})