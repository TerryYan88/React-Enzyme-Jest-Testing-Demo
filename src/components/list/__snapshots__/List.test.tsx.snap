// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should have a snapshot 1`] = `
.c0 {
  list-style-type: none;
}

<ul
  className="c0"
  data-test="ulComponent"
>
  <li
    className=""
    data-test="liComponent"
  >
    <h2
      data-test="componentTitle"
    >
      Example title
    </h2>
    <div
      data-test="componentDesc"
    >
      Some text
    </div>
  </li>
</ul>
`;
